Mathlib/RingTheory/Unramified/Pi.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.Unramified.Basic
+
+/-!
+
+# Formal-unramification of finite products of rings
+
+## Main result
+
+- `Algebra.FormallyUnramified.pi_iff`: If `I` is finite, `Π i : I, A i` is `R`-formally-smooth
+  if and only if each `A i` is `R`-formally-smooth.
+
+-/
+
+namespace Algebra.FormallyUnramified
+
+universe u v
+
+variable {R : Type max u v} {I : Type v} [Finite I] (f : I → Type max u v)
+variable [CommRing R] [∀ i, CommRing (f i)] [∀ i, Algebra R (f i)]
+
+theorem pi_iff :
+    FormallyUnramified R (∀ i, f i) ↔ ∀ i, FormallyUnramified R (f i) := by
```

Comment:
Do you have a strong reason for calling the family of algebras `f`? I would much prefer `A`, as in the module docstring. I also have a mild preference for `Π` over `∀` when working with non-props.